docs/book/src/concepts/machinepool.md

```diff
@@ -0,0 +1,92 @@
+- Matches provider IDs from the infrastructure resource with Kubernetes Nodes in the workload cluster.
+- Updates MachinePool status (ready replicas, conditions, etc.)
+
+## Why MachinePool?
+
+### Leverage provider primitives
+
+Most cloud providers already manage scaling, instance replacement, and health monitoring at the group level. MachinePool lets CAPI delegate lifecycle operations instead of duplicating that logic.
+
+**Example:**
+- AWS Auto Scaling Groups replace failed nodes automatically.
+- Azure VM Scale Sets support rolling upgrades with configurable surge/availability strategies.
+
+### Simplify upgrades and scaling
+
+Upgrades and scaling events are managed at the pool level:
+- Update Kubernetes version or bootstrap template → cloud provider handles rolling replacement.
+- Scale up/down replicas → provider adjusts capacity.
+
+This provides more predictable, cloud-native semantics compared to reconciling many individual Machine objects.
+
+### Autoscaling integration
+
+MachinePool integrates with the Cluster Autoscaler in the same way that MachineDeployments do. In practice, the autoscaler treats a MachinePool as a node group, enabling scale-up and scale-down decisions based on cluster load.
+
+### Tradeoffs and limitations
+
+While powerful, MachinePool comes with tradeoffs:
+
+- **Infrastructure provider complexity**: requires infrastructure providers to implement and maintain an InfrastructureMachinePool type.
+- **Less per-machine granularity**: you cannot configure each node individually; the pool defines a shared template.
+  > **Note**: While this is typically true, certain cloud providers do offer flexibility. 
+  > **Example**: AWS allows `AWSMachinepool.spec.mixedInstancesPolicy.instancesDistribution` while Azure allows `AzureMachinePool.spec.orchestrationMode`.
+- **Complex reconciliation**: node-to-providerID matching introduces edge cases (delays, inconsistent states).
+- **Draining**: The cloud resources for MachinePool may not necessarily support draining of Kubernetes worker nodes. For example, with an AWSMachinePool, AWS would normally terminate instances as quickly as possible. To solve this, tools like `aws-node-termination-handler` combined with ASG lifecycle hooks (defined in `AWSMachine.spec.lifecycleHooks`) must be installed, and is not a built-in feature of the infrastructure provider (CAPA in this example).
+- **Maturity**: The MachinePool API is still considered experimental/beta.
+
+## When to use MachinePool vs MachineDeployment
+
+Both MachineDeployment and MachinePool are valid options for managing worker nodes in Cluster API. The right choice depends on your infrastructure provider's capabilities and your operational requirements.
+
+### Use MachinePool when:
+
+- **Cloud provider supports scaling group primitives**: AWS Auto Scaling Groups, Azure Virtual Machine Scale Sets, GCP Managed Instance Groups. These resources natively handle scaling, rolling upgrades, and health checks.
```

Comment:
Should we perhaps mention OCI and Scaleway, the other two providers who support MachinePools? Also GCP isn't yet implemented so perhaps we shouldn't be listing it.